pkg/cmd/provisioning/alibabacloud/delete-ram-users.go

```diff
@@ -0,0 +1,123 @@
+	}
+
+	for _, credReq := range credRequests {
+		userName, _ := generateRAMUserName(fmt.Sprintf("%s-%s-%s", name, credReq.Namespace, credReq.Name))
+		listPoliciesReq := ram.CreateListPoliciesForUserRequest()
+		listPoliciesReq.UserName = userName
+		listPoliciesRes, err := client.ListPoliciesForUser(listPoliciesReq)
+		if err != nil {
+			return errors.Wrap(err, "Failed to list ram policies for component user")
+		}
+		//detach each policy from user
+		for _, userPolicy := range listPoliciesRes.Policies.Policy {
+			//detach component policy from the existing ram user
+			err := detachComponentPolicy(client, userPolicy.PolicyName, userName)
+			if err != nil {
+				return errors.Wrap(err, "Failed to detach ram policy from user")
+			}
+			//delete component ram policy
+			err = deleteComponentPolicy(client, userPolicy.PolicyName)
+			if err != nil {
+				return errors.Wrap(err, "Failed to delete component ram policy")
+			}
+		}
+		//delete component ram user
+		err = deleteComponentUser(client, userName)
+		if err != nil {
+			return errors.Wrap(err, "Failed to delete component user")
+		}
+	}
+	return nil
+}
+
+func deleteRAMUsersCmd(cmd *cobra.Command, args []string) {
+	envProvider := provider.NewEnvProvider()
+	profileProvider := provider.NewProfileProvider()
+	pc := provider.NewProviderChain([]provider.Provider{envProvider, profileProvider})
+	cred, err := pc.Resolve()
+	if err != nil {
+		log.Fatalf("Failed to create a client: %v", err)
+	}
+	clientConfig := sdk.NewConfig().WithScheme(apiRequestScheme)
+	client, err := alibabacloud.NewClient(DeleteRAMUsersOpts.Region, clientConfig, cred)
+	if err != nil {
+		log.Fatalf("Failed to create a client: %v", err)
+	}
+
+	err = deleteRAMUsers(client, DeleteRAMUsersOpts.Name, DeleteRAMUsersOpts.CredRequestDir)
+	if err != nil {
+		log.Fatalf("Failed to detach and delete the ram policy: %v", err)
+	}
+}
+
+// NewDeleteRAMUsersCmd provides the "delete-ram-users" subcommand
+func NewDeleteRAMUsersCmd() *cobra.Command {
+	detachCmd := &cobra.Command{
+		Use:   "delete-ram-users",
+		Short: "Detach RAM Policy from existing user",
+		Run:   deleteRAMUsersCmd,
+	}
+
+	detachCmd.PersistentFlags().StringVar(&DeleteRAMUsersOpts.Name, "name", "", "User-define name for all created Alibaba Cloud resources (can be separate from the cluster's infra-id)")
+	detachCmd.MarkPersistentFlagRequired("name")
+	detachCmd.PersistentFlags().StringVar(&DeleteRAMUsersOpts.CredRequestDir, "credentials-requests-dir", "", "Directory containing files of CredentialsRequests to create RAM AK for (can be created by running 'oc adm release extract --credentials-requests --cloud=alibabacloud' against an OpenShift release image)")
+	detachCmd.MarkPersistentFlagRequired("credentials-requests-dir")
+	detachCmd.PersistentFlags().StringVar(&DeleteRAMUsersOpts.Region, "region", "", "Alibaba Cloud region endpoint only required for GovCloud")
+	detachCmd.MarkPersistentFlagRequired("region")
```

Comment:
Is region really required for the RAM API calls made by the 'delete-ram-users' subcommand?

Reply:
removed